src/sources/socket/udp.rs

```diff
@@ -152,6 +166,41 @@ pub(super) fn udp(
                 })
             })?;
 
+        if !config.multicast_groups.is_empty() {
+            socket.set_multicast_loop_v4(true).unwrap();
+            let listen_addr = match config.address() {
+                SocketListenAddr::SocketAddr(SocketAddr::V4(addr)) => addr,
+                SocketListenAddr::SocketAddr(SocketAddr::V6(_)) => {
+                    // We could support Ipv6 multicast with the
+                    // https://doc.rust-lang.org/std/net/struct.UdpSocket.html#method.join_multicast_v6 method
+                    // and specifying the interface index as `0`, in order to bind all interfaces.
+                    panic!("IPv6 multicast is not supported")
```

Comment:
I thought about adding ipv6 support too... But as no one requested it yet, I thought i wouldn't be worth to implement & test it in this PR. Ayn thoughts about this? Should I include ipv6 support in this PR?

Reply:
Good call. Also, a separate PR is better because it make review easier. We can leave this as a potential improvement until there's demand for it. 

```suggestion
                    unimplemented!("IPv6 multicast is not supported")
```

Reply:
Totally agree!